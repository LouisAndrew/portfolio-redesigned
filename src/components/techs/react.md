---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
