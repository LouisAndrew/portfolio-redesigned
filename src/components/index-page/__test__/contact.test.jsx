import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Contact from '../contact'

describe('Contact component on landing page', () => {

        afterEach( cleanup )

        const headingList = [
                { h: 'Mock' },
                { h: 'Text' }
        ]

        const contactLists = [
                { icon: 'assets/mock.png', social: 'social-media', value: 'val1' },
                { icon: 'assets/mock2.png', social: 'social-media', value: 'val12', redirect: true }
        ]
        
        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render( <Contact />, div )
        })

        it('renders correctly', () => {

                const { getByTestId } = render( <Contact headingList={headingList} /> )
                const heading = getByTestId('con-heading').childNodes

                let i = 0
                heading.forEach( item => {

                        expect(item).toHaveTextContent( headingList[i].h )
                        i++
                } )
        })

        it('matches snapshot', () => {

                const tree = renderer.create( <Contact headingList={headingList} contactLists={contactLists} /> ).toJSON()
                expect(tree).toMatchSnapshot()
        })
})