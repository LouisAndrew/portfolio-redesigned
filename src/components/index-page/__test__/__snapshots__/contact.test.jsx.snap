// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component on landing page matches snapshot 1`] = `
<section
  className="sc-AxjAm eYetmd wrap"
  id="contact"
>
  <div
    className="sc-AxirZ ihFyRX"
    data-testid="con-heading"
  >
    <h1>
      Mock
    </h1>
    <h1>
      Text
    </h1>
  </div>
  <div
    className="sc-AxiKw WELlT"
    data-testid="con-list"
  >
    <div
      className="sc-AxhCb cuDfEf"
    >
      <img
        alt="social-media"
        src="assets/mock.png"
      />
      <p>
        val1
      </p>
    </div>
    <div
      className="sc-AxhCb cuDfEf"
    >
      <img
        alt="social-media"
        src="assets/mock2.png"
      />
      <a>
        val12
      </a>
    </div>
  </div>
</section>
`;
